### ISSUE TITLE
did something change with swaylock?

### ISSUE THREAD

**Issue by bbigras** [colemickens/nixpkgs-wayland#82]
I updated today and I saw that swayidle needs to be specified as an extraPackages now.

I wonder if it's the same with swaylock. I have `bindsym $mod+Print exec swaylock` in my config but I got 'command not found'.

If I add swaylock as an extraPackages I get the message:
`[shadow.c:19] swaylock needs to be setuid to read /etc/shadow`

**Comment by colemickens** [comment]
Yes, at the very least swaylock needs to be available via something like `extraPackages`. I do think `swaylock` had been working for a while in this configuration, since the split out, but I'm seeing the same thing.

I'm wondering if I'm missing a build dependency. I'm fixing some other breakage with latest updates and then will look into getting swaylock working.

Thanks for the report.

**Comment by colemickens** [comment]
(Also, I will update the README to show `swaylock` in `extraPackages` alongside `swayidle`.)

**Comment by bbigras** [comment]
Nice. Thanks!

**Comment by rileyrg** [comment]
installed from guix :  [source/shadow.c:19] swaylock needs to be setuid to read /etc/shadow

**Comment by berbiche** [comment]
@rileyrg 
> installed from guix : [source/shadow.c:19] swaylock needs to be setuid to read /etc/shadow

This isn't something that can we can fix unfortunately.

Packages in the nix store cannot be setuid/setgid.

How to achieve this with Guix is documented here: https://guix.gnu.org/manual/en/html_node/Setuid-Programs.html

**Comment by rileyrg** [comment]
> @rileyrg
> 
> > installed from guix : [source/shadow.c:19] swaylock needs to be setuid to read /etc/shadow
> 
> This isn't something that can we can fix unfortunately.
> 
> Packages in the nix store cannot be setuid/setgid.
> 
> How to achieve this with Guix is documented here: https://guix.gnu.org/manual/en/html_node/Setuid-Programs.html

Thanks for the reply. My google-fu was weak. I didnt find that link before.
